src/test/compile-fail/issue-12560-1.rs

```diff
@@ -8,15 +8,13 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-// compile-flags: -Z parse-only
-
 // For style and consistency reasons, non-parametrized enum variants must
 // be used simply as `ident` instead of `ident ()`.
 // This test-case covers enum declaration.
 
 enum Foo {
-    Bar(), //~ ERROR nullary enum variants are written with no trailing `( )`
-    Baz(), //~ ERROR nullary enum variants are written with no trailing `( )`
+    Bar(), //~ ERROR empty tuple structs and enum variants are not allowed
```

Comment:
I think this is pretty major regression to the error message. It was pretty clearly pointing out at the mistake before and telling how to fix it. Now it seems to suggest it is not allowed altogether, although only the syntax is incorrect.